stacks-signer/src/signerdb.rs

```diff
@@ -809,6 +812,20 @@ impl SignerDb {
             BlockState::try_from(state.as_str()).map_err(|_| DBError::Corruption)?,
         ))
     }
+
+    /// Return the all globally accepted block in a tenure (identified by its consensus hash).
+    pub fn get_globally_accepted_blocks(
+        &self,
+        tenure: &ConsensusHash,
+    ) -> Result<Vec<BlockInfo>, DBError> {
+        let query = "SELECT block_info FROM blocks WHERE consensus_hash = ?1 AND json_extract(block_info, '$.state') = ?2 ORDER BY stacks_height DESC";
```

Comment:
Would this include blocks with the same consensus hash, but from before the last tenure extend? i.e if a miner mines A, B, C, then D contains a tenure extend, then proposes E, I think we want this query to only return D, not A, B, and C.

Reply:
Yes you were right about what it returns. I just assumed it was easier to iterate and break rather than make the database do the work, but maybe I should?

Reply:
I just thought it might be easier to understand in the function rather than the database call especially in the case of no tenure change tx being found.

Reply:
Yeah, I think that's fine. If it becomes a bottleneck because we have so many blocks in one tenure, then we can consider another way.